### ISSUE TITLE
codigoProyecto

### ISSUE THREAD

**Issue by Josep0x400** [Josep0x400/ArduinoCode#6]
#include <HX711.h>
#include <LiquidCrystal_I2C.h>
#include <EEPROM.h>
#include <Servo.h>
#include <ESP8266WiFi.h>
#include <ESP8266HTTPClient.h>
#include <time.h>

#define PESO_MAX_MONEDA 49
#define PESO_MIN_MONEDA 12
#define SENSOR_CAP 2 //Amarillo
#define SCK A0 //Naranja
#define SRV 6 // Amarillo
#define DT A1 //Azul

LiquidCrystal_I2C lcd(0x27, 16, 2);
Servo serv;
HX711 bal;

bool captura;
int total = 0;
int pesa = 0;

const char* ssid = "Tu_SSID";          // Tu SSID de WiFi
const char* password = "Tu_Password";  // Tu contraseña de WiFi

const char* serverName = "https://mifirstproyect400.000webhostapp.com/CiberSeguridad/insertarValores.php";

// Función para obtener la fecha y hora actual
String getFormattedDate() {
  time_t now = time(nullptr);
  struct tm* timeinfo;
  char buffer[80];
  timeinfo = localtime(&now);
  strftime(buffer, 80, "%Y-%m-%d", timeinfo);
  return String(buffer);
}

String getFormattedTime() {
  time_t now = time(nullptr);
  struct tm* timeinfo;
  char buffer[80];
  timeinfo = localtime(&now);
  strftime(buffer, 80, "%H:%M:%S", timeinfo);
  return String(buffer);
}

void setup() {
  Serial.begin(38400);
  lcd.init();
  lcd.clear();
  lcd.backlight();
  lcd.begin(16, 2);
  lcd.print("Dame una moneda:");
  bal.begin(DT, SCK);
  bal.read();
  bal.set_scale(10);
  pinMode(SENSOR_CAP, INPUT_PULLUP);

  // Conexión a WiFi
  WiFi.begin(ssid, password);
  while (WiFi.status() != WL_CONNECTED) {
    delay(1000);
    Serial.println("Conectando a WiFi...");
  }
  Serial.println("Conectado a WiFi");

  // Inicialización de NTP para obtener la fecha y hora actual
  configTime(0, 0, "pool.ntp.org", "time.nist.gov");
  Serial.println("\nEsperando el tiempo NTP");
  while (!time(nullptr)) {
    Serial.print(".");
    delay(1000);
  }
  Serial.println("\nTiempo NTP obtenido");
}

void loop() {
  int giro = 180;
  captura = digitalRead(SENSOR_CAP);

  if (!captura) {
    bal.tare(2);
    delay(1500);
    pesa = bal.get_units(5) / 10;
    Serial.println(pesa);

    if (pesa < PESO_MIN_MONEDA || pesa > PESO_MAX_MONEDA) {
      lcd.clear();
      lcd.print("Esta no es una");
      lcd.setCursor(0, 1);
      lcd.print("moneda valida =(");
      giro = 0;
    }

    serv.attach(SRV);
    serv.write(90);
    delay(500);
    serv.write(giro);
    delay(500);
    serv.write(90);
    delay(500);
    serv.detach();

    if (pesa >= 12 && pesa <= 20) {
      total += 1;
      enviarDatosAServidor(1);
    } else if (pesa >= 21 && pesa <= 24) {
      total += 2;
      enviarDatosAServidor(2);
    } else if (pesa >= 25 && pesa <= 39) {
      total += 5;
      enviarDatosAServidor(5);
    } else if (pesa >= 40 && pesa <= 49) {
      total += 10;
      enviarDatosAServidor(10);
    } else {
      delay(2000);
      lcd.clear();
      lcd.setCursor(0, 0);
      lcd.print("Dame una moneda:");
    }

    lcd.setCursor(0, 1);
    lcd.print("Total:$");
    lcd.print(total);
  }
}

void enviarDatosAServidor(int valor) {
  if (WiFi.status() == WL_CONNECTED) {
    HTTPClient http;
    String fecha = getFormattedDate();
    String hora = getFormattedTime();
    String url = serverName + "?valor=" + String(valor) + "&fecha=" + fecha + "&hora=" + hora;

    http.begin(url);
    int httpResponseCode = http.GET();
    if (httpResponseCode > 0) {
      String payload = http.getString();
      Serial.println(payload);
    } else {
      Serial.print("Error en la solicitud HTTP: ");
      Serial.println(httpResponseCode);
    }
    http.end();
  } else {
    Serial.println("No conectado a WiFi");
  }
}
